### PR TITLE
LIVE-1468 Prevent useless rerenders on NFT Gallery

### PR DESCRIPTION
**Pull request by grsoares21**
This PR is purely technical and changes nothing feature-wise. It's sole objective is to improve the performance of the app while (and after) the user uses the NFT Gallery to see several NFTs.

It focuses mainly on preventing the useless rerendering on React components using standard techniques such as:
* Moving objects, arrays and lambda function declarations outside of the render function
* Memoizing expensive components
* Splitting and memoizing components selecting only parts of contexts

The improvements are seen in the following measurements.
**Before:**
https://user-images.githubusercontent.com/6013294/156390038-dee32df0-e18d-4638-96d1-8e0510f78928.mp4
profiler-nft-without-fix

**After:**
https://user-images.githubusercontent.com/6013294/156390061-d0dfc244-023c-4186-9565-f985d8be35cc.mp4
profiler-nft-with-fix

### Type
UX Improvement

### Context
https://ledgerhq.atlassian.net/browse/LIVE-1468

### Parts of the app affected / Test plan
Use an account with several NFTs, go to the NFT Gallery page and scroll down to load several NFTs. The app should remain snappy and responsive while and after these NFTs are loaded.

### CODE REVIEW

File: src/components/Button.js [hunk 1]
```diff
@@ -29,7 +29,6 @@ const ANIM_OFFSET = 20;
 const ANIM_DURATION = 300;
 
 type LTextProps = React$ElementProps<typeof LText>;
-type LTextStyleProp = $PropertyType<LTextProps, "style">;
```
Comment (juan-cortes) [inline src/components/Button.js:32]: lint

File: src/components/OperationRow.js [hunk 2]
```diff
@@ -288,3 +288,7 @@ const styles = StyleSheet.create({
     justifyContent: "center",
   },
 });
+
+const operationRowNftNameStyles = [styles.bodyRight, { maxWidth: "50%" }];
+
+export default memo<Props>(OperationRow);
```
Comment (juan-cortes) [inline src/components/OperationRow.js:294]: prettier

**Review by juan-cortes — CHANGES_REQUESTED** [review]
Looks good to me, I trust you on the performance gains, just clean it up 👍🏼

**Review by juan-cortes — APPROVED** [review]

**Review by ychen-ledger — APPROVED** [review]
Validated with: 

- Pixel 4 Android 12
- iPhone 11 iOS 14.7
- Gaspard account

**Review by ychen-ledger — APPROVED** [review]
Validated with: 

- Pixel 4 Android 12 RAM 6GB
- iPhone 11 iOS 14.7 RAM 4GB
- Gaspard account
- On an old Android phone(ex: Pixel 3a Android 12 RAM 4 GB), app can happen with crash on Gaspard account or slow on around 100 NFTs account

**Pull request merged**